libnavui-voice/src/test/java/com/mapbox/navigation/ui/voice/api/TimeBasedNextVoiceInstructionsProviderTest.kt

```diff
@@ -0,0 +1,752 @@
+package com.mapbox.navigation.ui.voice.api
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.LegStep
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.api.directions.v5.models.VoiceInstructions
+import com.mapbox.navigation.testing.LoggingFrontendTestRule
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class TimeBasedNextVoiceInstructionsProviderTest {
```

Comment:
What do you think about having more specific tests like this:
```
val route = loadRealRouteFromJson()
val instructionsToDownload = provider.getNextVoiceInstructions(/* provide some specific place on the route */)
assertEquals(listOf("instruction 1", "instruction 2"), instructionsToDownload.map { it.announcement }
```

this would demonstrate how this work on read data/cases. What do you think about it?

Reply:
This approach would not allow us to test corner cases. And there are a loooot of them. I've changed the code 5+ times when I was writing these tests and these changes were improvements. 
With the real data you only cover the very base scenarios. And also with these huge jsons it's very hard to fix tests when they fail later.
It's a unit test so I'm in favour of simple data with lots of corner cases. Sanity check with big data can be performed in either instrumentation tests or when running examples.

Reply:
Just in case, I'm not trying to say that it's something wrong with existing tests, they are great! 👍

I think that one test with real data would be a great addition to exiting ones. It would prove that everything works on the real data as well. What's more, it could be used as an example of how the feature works: "if you're here on this route the feature downloads following voice instructions". It's similar to how I would test it manually on examples, but fast 🙂 

It's okay to test with real data in instrumentation tests, but there is nothing android specific, why not to test in on fast JVM? 🤔 

Anyway, it's not a strong opinion, just a suggestion. If you ignore it, I'm fine with it. Current tests suite is great already 👍